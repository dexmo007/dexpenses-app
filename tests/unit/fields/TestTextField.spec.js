import Vue from 'vue';
import Vuetify from 'vuetify';
import { mount } from '@vue/test-utils';

Vue.use(Vuetify, {}); // should actually use localVue, but that causes a console error atm

describe('Test text field', () => {
  it('should work', done => {
    const wrapper = mount({
      template: `<v-text-field @input="$emit('input', {data: $event})" mask="##/##" />`,
    });
    wrapper.find('input').setValue('12/34');
    expect(wrapper.element).toMatchSnapshot();
    wrapper.vm.$nextTick(() => {
      expect(wrapper.emitted().input).toBeTruthy();
      done();
    });

    // console.log(wrapper.emitted().input);
  });
});
