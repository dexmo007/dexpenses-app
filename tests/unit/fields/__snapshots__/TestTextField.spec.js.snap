// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test text field should work 1`] = `
<div
  class="v-input v-text-field theme--light"
>
  <div
    class="v-input__control"
  >
    <div
      class="v-input__slot"
    >
      <div
        class="v-text-field__slot"
      >
        <input
          maxlength="5"
          type="text"
        />
      </div>
    </div>
    <div
      class="v-text-field__details"
    >
      <div
        class="v-messages theme--light"
      >
        <div
          class="v-messages__wrapper"
        />
      </div>
    </div>
  </div>
</div>
`;
